File: 100daysofcode/0283-move-zeroes/solution.cpp
class Solution {
public:
    vector<int> moveZeroes(vector<int>& nums) {
        int n=nums.size();
        int l=-1;
        for(int i=0;i<n;i++){
            if(nums[i]==0){
                l=i;
                break;
            }
        }
        if(l==-1) return nums;
        for(int j=l+1;j<n;j++){
            if(nums[j]!=0){
                swap(nums[j],nums[l]);
                l++;
            }
        }
        return nums;
    }
};

File: 100daysofcode/0283-move-zeroes/solution.java
class Solution {
    public int[] moveZeroes(int[] nums) {

        int[] ans = new int[nums.length];
        int end = 1;
        int start = 0;

        for(int i = 0; i< nums.length; i++){
            if(nums[i] == 0){
                ans[nums.length - end] = 0;
                end++;

            }
            else{
                ans[start] = nums[i];
                start++;
            }
        }
        for(int i = 0 ; i< nums.length; i++){
            nums[i] = ans [i];
        }
        return nums;
        
    }
}
